scripts/py_matter_yamltests/matter_yamltests/parser.py

```diff
@@ -698,6 +699,20 @@ def event_number(self, value):
     def pics(self):
         return self._test.pics
 
+    def _get_last_event_number(self, responses) -> Optional[int]:
+        if not self.is_event:
+            return None
+
+        last_event = responses[-1]
+        if not isinstance(last_event, dict):
+            return None
```

Comment:
I wonder if you want to iterate over the responses. We do have a few commands that may returns multiple responses that are not all of the same types (thinking of `./out/debug/standalone/chip-tool any read-all`) so it could be that the last response is not an event despites having some event numbers in the response.